### PR TITLE
test(serialize): additional tests for name, domain and path RFC validations

### PR DESCRIPTION
**Pull request by hdtmccallie**
These tests better align with the proper RFC rules for the cookie attributes of `name`, `domain`, and `path`.

These test are meant to be implemented along with [PR #167][1] that adds more fine-grained validations based on RFC rules.

[1]: https://github.com/jshttp/cookie/pull/167

### CODE REVIEW

File: test/serialize.js [hunk 1]
```diff
@@ -17,24 +17,98 @@ describe('cookie.serialize(name, value)', function () {
...
+      'foo\n',
+      'foo\u280a',
+      'foo/foo',
+      'foo,foo',
+      'foo;foo',
+      'foo@foo',
+      'foo[foo]',
+      'foo?foo',
+      'foo:foo',
+      'foo!foo',
```
Comment (blakeembrey) [inline test/serialize.js:55]: This was in the valid name list.

File: test/serialize.js [hunk 2]
```diff
@@ -17,24 +17,98 @@ describe('cookie.serialize(name, value)', function () {
...
+      'foo.bar',
+      'foo^bar',
+      'foo_bar',
+      'foo`bar',
+      'foo|bar',
+      'foo~bar',
+      'foo7bar',
+    ];
+
+    validNames.forEach((name) => {
```
Comment (blakeembrey) [inline test/serialize.js:39]: This project runs all the way back to node 0.x so needed to avoid the new syntax.

File: test/serialize.js [hunk 3]
```diff
@@ -133,14 +207,47 @@ describe('cookie.serialize(name, value, options)', function () {
...
 
     it('should throw for invalid value', function () {
-      assert.throws(cookie.serialize.bind(cookie, 'foo', 'bar', { path: '/\n' }),
-        /option path is invalid/)
-    })
-  })
+      const invalidPaths = [
+        '/\n',
+        '/foo\u0000',
+        '/foo bar',
```
Comment (blakeembrey) [inline test/serialize.js:235]: This wasn't invalid, should it be? The spec says `<any CHAR except CTLs or ";">` so I believe space is fine?

File: test/serialize.js [hunk 4]
```diff
@@ -133,14 +207,47 @@ describe('cookie.serialize(name, value, options)', function () {
...
+        '/path/with\rnewline',
+        '/path\\with\\backslash',
```
Comment (blakeembrey) [inline test/serialize.js:237]: Same with a backslash, also fine according to `<any CHAR except CTLs or ";">`.

File: test/serialize.js (hunk 1 above)
Comment (hdtmccallie) [reply test/serialize.js:55]: You are correct, good catch.

File: test/serialize.js (hunk 2 above)
Comment (hdtmccallie) [reply test/serialize.js:39]: Copy that. I had it pinned with 0.12 for the install and tests but was running 8 for debug so missed this.

File: test/serialize.js (hunk 3 above)
Comment (hdtmccallie) [reply test/serialize.js:235]: You are correct. Space should be acceptable.

File: test/serialize.js (hunk 4 above)
Comment (hdtmccallie) [reply test/serialize.js:237]: Correct as well.

**Pull request merged**